src/Random123.jl

```diff
@@ -27,15 +27,38 @@ include("threefry.jl")
 export Philox2x, Philox4x
 include("philox.jl")
 
-if R123_USE_AESNI
-    include("aesni_common.jl")
-    export AESNI1x, AESNI4x
-    include("aesni.jl")
+"True when AES-NI has been enabled."
+const R123_USE_AESNI = Ref(false)
+export R123_USE_AESNI
+export AESNI1x, AESNI4x
+export ARS1x, ARS4x
+include("./aesni/module.jl")
 
-    export ARS1x, ARS4x
-    include("ars.jl")
-else
-    @warn "AES-NI is not enabled, so AESNI and ARS are not available."
+function __init__()
+
+    R123_USE_AESNI[] = try
+        cmd = Base.julia_cmd()
+        push!(
+            cmd.exec, "-e",
+            "const __m128i = NTuple{2, VecElement{UInt64}};" *
+            "@assert ccall(\"llvm.x86.aesni.aeskeygenassist\", " *
+            "llvmcall, __m128i, (__m128i, UInt8), " *
+            "__m128i((0x0123456789123450, 0x9876543210987654)), 0x1) ≡ " *
+            "__m128i((0x857c266f7c266e85, 0x2346382146382023))"
+        )
+        success(cmd)
+    catch e
+        @show e
+        false
+    end
+    
+    if R123_USE_AESNI[]
+        @eval using ._AESNIModule
```

Comment:
This is not allowed, and triggers https://github.com/JuliaGPU/CUDA.jl/issues/1422.

Reply:
I have seen that. I realized `@eval` may not be necessary. Will look into this later..

Reply:
You cannot do a `using` at non-toplevel scope. So this will never work from `__init__`.

Reply:
Oh I see... Is there a way to lazy load this? Like what Requires.jl does

Reply:
It's possible, one part of the trick is not to do so while precompiling, https://github.com/JuliaPackaging/Requires.jl/blob/542a641b633fa38a5950c2b1dbb90861967c7015/src/require.jl#L97-L105.